### PR TITLE
enhancement(redis sink): Adhere to instrumentation spec

### PR DESCRIPTION
**Pull request by neuronull**
Closes:  #14141

Epic: https://github.com/vectordotdev/vector/issues/13995

- Added comments where common code is handling Errors
- Removed double emission in Error path
- Added compliance test helpers to remaining tests/int tests

### CODE REVIEW

File: src/internal_events/redis.rs [hunk 1]
```diff
@@ -3,43 +3,6 @@ use vector_core::internal_event::InternalEvent;
 
 use vector_common::internal_event::{error_stage, error_type};
 
-#[derive(Debug)]
-pub struct RedisSendEventError<'a> {
-    error: &'a redis::RedisError,
-    error_code: String,
-}
```
Comment (spencergilbert) [inline src/internal_events/redis.rs:10]: Duplicate from underlying sink framework?

**Review by spencergilbert — APPROVED** [review]
LGTM other than the clippy ding

File: src/internal_events/redis.rs (hunk 1 above)
Comment (neuronull) [reply src/internal_events/redis.rs:10]: Correct - specifically `SinkSendError` in the retry logic.

The reason I'm adding the following comment to every sink I audit that this applies to, is to hopefully prevent future attempts to add Error emission in the service call function:

```
// Emission of Error internal event is handled upstream by the caller
```

Comment (spencergilbert) [reply src/internal_events/redis.rs:10]: Just making sure, tying your comments in the PR to the changes 😄

Comment (neuronull) [reply src/internal_events/redis.rs:10]: I probably should have mentioned that in the comment 0, next time.

Comment (spencergilbert) [reply src/internal_events/redis.rs:10]: Meh, it seemed obvious - I just wanted to double check

**Pull request merged**